doc/design/session.md

```diff
@@ -0,0 +1,62 @@
+# Design Doc: Session
+
+## Abstract
+
+This design doc proposes to have an object called *Session* which
```

Comment:
Is these design came out from https://github.com/PaddlePaddle/Paddle/pull/3811#issuecomment-328180377?

If so, you probably need to add descriptions like "session is able to distinguish running a graph locally or remotely, using CPU or GPU, using one device or more"

Reply:
Great point! Added "The Session is able to distinguish running a graph locally or remotely, using CPU only or using one or more GPUs."